### ISSUE TITLE
Build Failure: Apple Silicon - “C++ linker does not work with specified blank $LIBS and specified $LINKFLAGS”

### ISSUE THREAD

**Issue by kitelessd** [dxx-rebirth/dxx-rebirth#570]
### Environment
Downloaded source from Github. Installed latest version of HomeBrew using directions from their web site. 
Installed all prerequisites from DXX Markdown instructions. Trying to build and place in sub-dir. 

#### Operating System Environment
Mac OS Big Sur 11.2, Homebrew

#### CPU environment
Apple Silicon M1 (Mac Mini) 

#### Issue
When compiling based off markdown instructions I get this error: 
`C++ linker does not work with specified blank $LIBS and specified $LINKFLAGS`

When using "macos_add_frameworks=0" arg I get this error: 
`Header physfs.h is missing or includes a missing supporting header.`

### Steps to Reproduce
Please see my original post that I put in the wrong thread over here: https://github.com/dxx-rebirth/dxx-rebirth/issues/503#issuecomment-774321330

Per the response from @vKLp here is my original issue below. 

At the time I had not seen that people were using the "macos_add_frameworks=0" flag. So this is the command I was using originally: 

% scons sdlmixer=1 d1x=1 builddir=dx1-build

The run ends in this error: 

`scons: *** C++ compiler works.  C++ linker does not work with specified blank $LIBS and specified $LINKFLAGS.  C++ linker does not work with blank $LIBS and blank $LINKFLAGS.  See dx1-build/sconf.log for details.  Stop.`

sconf_linker.log attached to this message. 

Executing this command based on the "Mac OS X build notes: here on Github:

`% scons opengl=1 sdlmixer=1 d1x=1 macos_add_frameworks=0 builddir=dx1-build`

`scons: *** Header physfs.h is missing or includes a missing supporting header.  See dx1-build/sconf.log for details.  Stop.`

sconf_physfs.log also attached to this message. 

Note: I'd like to compile Dxx1 and 2 but just trying 1 for these compile attempts. Thanks for any help, this isn't my profession so I'm kind of fumbling my way through this. Would be happy to post the .app compiled versions of these for other Apple Silicon users to download and use. Would post without copyrighted game data of course. :) 

[sconf_linker.log](https://github.com/dxx-rebirth/dxx-rebirth/files/5937794/sconf_linker.log)
[sconf_physfs.log](https://github.com/dxx-rebirth/dxx-rebirth/files/5937795/sconf_physfs.log)

**Comment by vLKp** [comment]
As a general Github tip, you need to spell my username correctly for the system to recognize it as a reference to me. ;)

`sconf_linker.log`: you do not have an SDL framework installed.  If you installed SDL as just the traditional C headers / libraries, rather than wrapped in the Apple-specific framework concept, then this is expected.  Using `macos_add_frameworks=0` to bypass it is fine, with the caveat that if you suppress the frameworks, then it is your responsibility to have all the required headers / libraries in the appropriate search paths, and you may need to add some libraries to `LIBS` that would otherwise have been handled automatically by the default `FRAMEWORKS` value.  We can come back to that when you get to the point of linker errors.

`sconf_physfs.log`: the compiler could not find `physfs.h` in any of the standard search paths.  How exactly did you install `physfs` on the system?  I would have expected its headers to be in `/usr/local/include`, and that your compiler would search there.  What is the output of `ls -la /usr/local/include/ /usr/local/lib/`?  What does `brew` tell you about the physfs that it installed?  (Sorry, I don't know the command to query brew's state.)

**Comment by kitelessd** [comment]
Thanks for the quick response. Dang proofread 3 times and still missed that I got your name wrong. haha. 

OK I think we (a Mac coder friend and I) have this worked out but it graduated to a different problem. First of all, the solution. The "smoking gun" is here: https://docs.brew.sh/Installation
tl;dr Intel Homebrew installs here: "/usr/local" and Apple Silicon here: "/opt/homebrew"

I installed SDL & SDL2 via Homebrew and confirmed they are installed. Also you will find this funny probably, 

```
% pkg-config --cflags --libs physfs
-I/opt/homebrew/Cellar/physfs/3.0.2/include -L/opt/homebrew/Cellar/physfs/3.0.2/lib -lphysfs
```

But now it seems obvious what the issue here is for AS builds. So the bandaid solution: 

`sudo ln -s /opt/homebrew/Cellar/physfs/3.0.2/include/* /usr/local/include && sudo ln -s /opt/homebrew/Cellar/physfs/3.0.2/lib/* /usr/local/lib`

After creating the dirs "/usr/local/include" and "/usr/local/lib" I made the symlinks and bam! It compiled... well till it stopped.

Log attached. It may be this problem is already covered someplace else but I haven't looked yet. Also I might be the first person to try and compile this for Apple Silicon (AS), so hopefully this will be helpful in the future. 
 
Command: `scons opengl=1 sdlmixer=1 d1x=1 macos_add_frameworks=0 builddir=dx1-build`

Main error in terminal window but I'm sure this is in the log too. 
```
Similar/main/game.cpp:190:23: error: unused variable 'initial_small_game_screen_mode' [-Werror,-Wunused-const-variable]
constexpr screen_mode initial_small_game_screen_mode{320, 200};
```

[sconf.log](https://github.com/dxx-rebirth/dxx-rebirth/files/5937876/sconf.log)

**Comment by vLKp** [comment]
For most packages, if `pkg-config` can find it, then SConstruct should automatically find it.  However, looking at SConstruct, that is not true for physfs.  SConstruct assumes that no call to `pkg-config` is needed.  I think this should be changed to be consistent with the other packages.  I pushed 791282aad77a87fd2f3c86718a4069ea81a6172c to make this change.  Please sync, remove your symlinks from `/usr/local/include`, and retry.

**Comment by vLKp** [comment]
That error with the unused variable was reported as a comment on an unrelated issue earlier today, and is fixed in the latest code.

**Comment by kitelessd** [comment]
Just wanted to confirm that this builds properly now after your update. This is the latest build released today with your latest fixes. 

sconf.log build log attached. Closing ticker. Thanks so much for keeping this project alive! Wild being able to play Descent on a mainstream non-x86/64 architecture!  

[sconf.log](https://github.com/dxx-rebirth/dxx-rebirth/files/5941242/sconf.log)


command: scons opengl=1 sdlmixer=1 d1x=1 macos_add_frameworks=0 builddir=dx1-build
